gef-ui/server/controllers/unissued-facilities/index.js

```diff
@@ -0,0 +1,516 @@
+  }
+
+  if (issueDateIsFullyComplete && coverStartDateIsFullyComplete) {
+    if (coverStartDate < issueDate) {
+      aboutFacilityErrors.push({
+        errRef: 'coverStartDate',
+        errMsg: 'Cover start date cannot be before the issue date',
+      });
+    }
+  }
+
+  // Regex tests to see if value is a number only
+  const digitsRegex = /^[0-9]*$/;
+  if (body.monthsOfCover) {
+    if (!digitsRegex.test(body.monthsOfCover)) {
+      aboutFacilityErrors.push({
+        errRef: 'monthsOfCover',
+        errMsg: 'You can only enter numbers',
+      });
+    }
+    if (body.monthsOfCover > 999) {
+      aboutFacilityErrors.push({
+        errRef: 'monthsOfCover',
+        errMsg: 'You can only enter a maximum of 999 months cover',
+      });
+    }
+  }
+
+  if (aboutFacilityErrors.length > 0) {
+    return res.render('partials/unissued-change-about-facility.njk', {
+      errors: validationErrorHandler(aboutFacilityErrors),
+      facilityName: body.facilityName,
+      shouldCoverStartOnSubmission: body.shouldCoverStartOnSubmission,
+      monthsOfCover: body.monthsOfCover,
+      hasBeenIssued: true,
+      issueDateDay,
+      issueDateMonth,
+      issueDateYear,
+      coverStartDateDay,
+      coverStartDateMonth,
+      coverStartDateYear,
+      coverEndDateDay,
+      coverEndDateMonth,
+      coverEndDateYear,
+      facilityType,
+      facilityTypeString,
+      dealId,
+      facilityId,
+      status,
+    });
+  }
+
+  return {
+    issueDate,
+    coverStartDate,
+    coverEndDate,
+    body,
+    facilityId,
+    dealId,
+  };
+};
+
+/**
+ * post for changing unissued facilities to issued from unissued facilities list
+ * validates first and gets parameters from validation
+ * displays success message and redirects to unissued facilities list
+ */
+const postChangeUnissuedAboutFacility = async (req, res) => {
+  const {
+    issueDate, coverStartDate, coverEndDate, body, facilityId, dealId,
+  } = await facilityValidation(req, res);
+
+  try {
+    await api.updateFacility(
+      facilityId,
+      {
+        name: body.facilityName,
+        shouldCoverStartOnSubmission: isTrueSet(body.shouldCoverStartOnSubmission),
+        monthsOfCover: body.monthsOfCover || null,
+        issueDate: issueDate ? format(issueDate, CONSTANTS.DATE_FORMAT.COVER) : null,
+        coverStartDate: coverStartDate ? format(coverStartDate, CONSTANTS.DATE_FORMAT.COVER) : null,
+        coverEndDate: coverEndDate ? format(coverEndDate, CONSTANTS.DATE_FORMAT.COVER) : null,
+        hasBeenIssued: true,
+        changedToIssued: true,
+        coverDateConfirmed: true,
+      },
+      (req.success = {
+        message: `${body.facilityName} is updated`,
+      }),
+      (req.url = `/gef/application-details/${dealId}/unissued-facilities`),
+    );
+    // TODO: redirect where it changes the link away from /gef/application-details/${dealId}/unissued-facilities/facilityid
+    //  ... to /gef/application-details/${dealId}/unissued-facilities
```

Comment:
Is this something that needs doing? Should we create jira ticket? Want to avoid TODO comments without jira numbers 😉

Reply:
needs to be done as refresh causes resubmit of form.  Added jira.  Will be big job to do now and is not breaking anything